internal/controller/drplacementcontrol.go

```diff
@@ -1890,64 +1890,85 @@ func (d *DRPCInstance) newVRGSpecSync() *rmn.VRGSyncSpec {
 	}
 }
 
-func dRPolicySupportsMetro(drpolicy *rmn.DRPolicy, drclusters []rmn.DRCluster) (bool, map[string][]string) {
-	syncPeerClasses := drpolicy.Status.Sync.PeerClasses
-	aSyncPeerClasses := drpolicy.Status.Async.PeerClasses
-
-	if len(syncPeerClasses) == 0 && len(aSyncPeerClasses) == 0 {
+// dRPolicySupportsMetro returns a boolean indicating that the policy supports a Metro(Sync) DR capability, and a
+// list of list of strings, where each list of strings contains the name of the clusters that are in a Sync
+// relationship
+func dRPolicySupportsMetro(drpolicy *rmn.DRPolicy, drclusters []rmn.DRCluster, drClusterIDsToNames map[string]string) (
+	supportsMetro bool,
+	metroClustersInPolicy [][]string,
+) {
+	if len(drpolicy.Status.Sync.PeerClasses) == 0 && len(drpolicy.Status.Async.PeerClasses) == 0 {
 		return checkMetroSupportUsingRegion(drpolicy, drclusters)
 	}
 
-	if len(syncPeerClasses) != 0 {
-		return checkMetroSupportUsingPeerClass(drpolicy)
+	if len(drpolicy.Status.Sync.PeerClasses) != 0 {
+		return checkMetroSupportUsingPeerClass(drpolicy, drClusterIDsToNames)
 	}
 
 	return false, nil
 }
 
-func checkMetroSupportUsingPeerClass(drPolicy *rmn.DRPolicy) (bool, map[string][]string) {
-	peerClasses := drPolicy.Status.Sync.PeerClasses
-	if len(peerClasses) == 0 {
+func checkMetroSupportUsingPeerClass(drPolicy *rmn.DRPolicy, drClusterIDsToNames map[string]string) (bool, [][]string) {
+	if len(drPolicy.Status.Sync.PeerClasses) == 0 {
 		return false, nil
 	}
 
-	metroMap := make(map[string][]string)
-	for _, peerClass := range peerClasses {
-		metroMap[peerClass.StorageClassName] = peerClass.ClusterIDs
+	if drClusterIDsToNames == nil {
+		return true, nil
```

Comment:
Should this be true, nil or false, nil?

Reply:
It should be true, nil since we want to process further only when ClusterIDToNames are present and these are present only when validatingConflicts. we can just return true,nil  in all other cases where we just want to check if Metro is true or not.

Reply:
The previous check is sufficient to determine that Sync is supported, so when callers do not pass in the ID map, the answer is true and a nil return clusters list. So I am stating this is correct as is.